MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs

```diff
@@ -1155,10 +1155,13 @@ public IReadOnlyList<string> GetPrimaryPlaylistM2tsFiles(string path)
 
             // Get all files from the BDMV/STREAMING directory
             // Only return playable local .m2ts files
+            var directoryFiles = _fileSystem.GetFiles(Path.Join(path, "BDMV", "STREAM")).ToList();
```

Comment:
This can be simplified (may contain syntax errors)
```C#
            var directoryFiles = _fileSystem.GetFiles(Path.Join(path, "BDMV", "STREAM"), [".mt2s"], true)
                .Where(f => validPlaybackFiles.Contains(Path.GetFileName(f.Fullname.AsSpan()), StringComparison.OrdinalIgnoreCase))
                .Select(f => f.FullName);

Reply:
Yes but iterate over result of GetFIles may broke the order of playable files(Not 100% sure but I believe it's what [this commit](https://github.com/jellyfin/jellyfin/commit/6777f47e0e00b22562061c3b53e6aa923449fb70) trying to do).

Reply:
Then this
```C#
            var files = _fileSystem.GetFiles(Path.Join(path, "BDMV", "STREAM"), new[] { ".mt2s" }, true, false).ToList();
            return validPlaybackFiles
                .Select(validFile => files.FirstOrDefault(f => Path.GetFileName(f.FullName.AsSpan()).Equals(validFile, StringComparison.OrdinalIgnoreCase))?.FullName)
                .Where(f => f is not null)
                .ToList();

Reply:
You are the best!